### PR TITLE
[Tests]subresource_api_test.go:Remove use of StartVirtualMachine func

### PR DESCRIPTION
**Pull request by Barakmor1**
StartVirtualMachine should be deprecated as it is part of tests/utils.go and use expensive get and update API calls

Instead of Updating VM runStrategy in subresource_api_test.go create vm with with the desired runStrategy.

### What this PR does
Before this PR:

After this PR:

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

**Review by fossedihelm — APPROVED** [review]

**Review by EdDev — APPROVED** [review]

**Review by enp0s3 — COMMENTED** [review]
/approve

File: tests/subresource_api_test.go [hunk 1]
```diff
@@ -161,15 +161,13 @@ var _ = Describe("[sig-compute]Subresource Api", decorators.SigCompute, func() {
...
-				vm := libvmi.NewVirtualMachine(libvmifact.NewGuestless())
+				vm := libvmi.NewVirtualMachine(libvmifact.NewGuestless(), libvmi.WithRunStrategy(v1.RunStrategyAlways))
 				vm, err := virtCli.VirtualMachine(testsuite.GetTestNamespace(vm)).Create(context.Background(), vm, metav1.CreateOptions{})
 				Expect(err).NotTo(HaveOccurred())
 
-				tests.StartVirtualMachine(vm)
+				Eventually(matcher.ThisVM(vm)).WithTimeout(300 * time.Second).WithPolling(time.Second).Should(matcher.BeReady())
 				vmi, err := virtCli.VirtualMachineInstance(testsuite.GetTestNamespace(vm)).Get(context.Background(), vm.Name, metav1.GetOptions{})
 				Expect(err).NotTo(HaveOccurred())
-				Expect(vmi.Status.Phase).To(Equal(v1.Running))
```
Comment (xpivarc) [inline tests/subresource_api_test.go:171]: Why was this removed?

**Review by xpivarc — COMMENTED** [review]
/hold
We should deprecate this function first if nobody is committing to remove this as whole. This will come with explicit approval that we want to do something like this and it will make sure we converge.

**Comment by Barakmor1** [comment]
I don’t believe this is a strong reason to hold up the PR. However, here is the PR for the deprecation:
https://github.com/kubevirt/kubevirt/pull/12572

/unhold

File: tests/subresource_api_test.go (hunk 1 above)
Comment (Barakmor1) [reply tests/subresource_api_test.go:171]: https://github.com/kubevirt/kubevirt/blob/ec14523ae05a7053516b0a8bd2e229cf9d96500c/staging/src/kubevirt.io/api/core/v1/types.go#L1625

IIUC we're waiting for it to be ready, it means it's already running, so there's no need to check it again.

Comment (xpivarc) [reply tests/subresource_api_test.go:171]: @Barakmor1 
The cost of the check is limit to zero. On the other hand if we ever break the API contract this check would catch it. Consider re-adding it.
